tasks/release.py

```diff
@@ -0,0 +1,74 @@
+# -*- coding: utf-8 -*-
+"""Handles creating a release PR"""
+import sys
+from pathlib import Path
+from subprocess import check_call
+from typing import Tuple
+
+from git import Commit, Head, Remote, Repo, TagReference
+from packaging.version import Version
+
+ROOT_SRC_DIR = Path(__file__).parents[1]
+
+
+def main(version_str: str) -> None:
+    version = Version(version_str)
+    repo = Repo(str(ROOT_SRC_DIR))
+
+    if repo.is_dirty():
+        raise RuntimeError("Current repository is dirty. Please commit any changes and try again.")
+
+    upstream, release_branch = create_release_branch(repo, version)
+    release_commit = release_changelog(repo, version)
+    tag = tag_release_commit(release_commit, repo, version)
+    print("push release tag")
```

Comment:
If the CI fails, we abort this release altogether as a botched release, and try again?

Reply:
Mostly, I personally would go for remove the version tag. Fix the CI; create a new PR against master with the fixes. Restart.